metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -32,11 +33,25 @@ object CompletionProvider extends LazyLogging {
         isUsedLabel += label
         items += CompletionItem(
           label = label,
-          detail = Some(r.sym.signatureString)
+          detail = Some(r.sym.signatureString),
+          kind = symbolKind(compiler)(r)
         )
       }
     }
     CompletionList(isIncomplete = false, items = items.result())
   }
 
+  private def symbolKind(c: Global)(r: c.CompletionResult#M): Option[Int] = {
+    if (r.sym.isConstructor) Some(SymbolKind.Constructor)
+    else if (r.sym.isModuleOrModuleClass) Some(SymbolKind.Module)
+    else if (r.sym.isTraitOrInterface) Some(SymbolKind.Interface)
+    else if (r.sym.isClass) Some(SymbolKind.Class)
+    else if (r.sym.isPackageObject) Some(SymbolKind.Namespace)
+    else if (r.sym.isPackage) Some(SymbolKind.Package)
+    else if (r.sym.isMethod) Some(SymbolKind.Method)
+    else if (r.sym.isVal) Some(SymbolKind.Constant)
+    else if (r.sym.isVar) Some(SymbolKind.Variable)
```

Comment:
Note that `CompletionItemKind` and `SymbolKind` are different things and even same names correspond to different numbers 💣  I think you should use `CompletionItemKind` here. It's already defined [here](https://github.com/scalameta/language-server/blob/master/languageserver/src/main/scala/langserver/types/types.scala#L89), but is by the way incomplete in respect to the v3 of the LSP spec.

Reply:
Argh, thanks, that's a huge mistake I've made. A newtype would've prevented this :(

Reply:
And I think this is one thing we should change in that code: there's no point to copy the untyped spec style with "named number" if we can have something more Scala-idiomatic (I wanted to say enums, but 😅  you know..)

Reply:
Also, I found https://github.com/Microsoft/language-server-protocol/commit/05e3af34464671a788679080f06558db52d1097b

In version 4 of the protocol we'll be able to specify which symbols we support :)

Reply:
> And I think this is one thing we should change in that code: there's no point to copy the untyped spec style with "named number" if we can have something more Scala-idiomatic

Yes, and ADT could be a better choice here, or even just a value class.

Reply:
> In version 4 of the protocol we'll be able to specify which symbols we support :)

I thought it's included in v3 (not released yet), isn't it?

Reply:
Btw, in that referred commit, a special symbol kind for `null` made me laugh :rofl: